app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -3204,56 +3308,311 @@ export function testsForRpcMethodSupportingBlockParam(
         });
       });
 
-      it('makes an additional request to the RPC endpoint if the given block number matches the latest block number', async () => {
-        await withMockedCommunications({ providerType }, async (comms) => {
-          const request = {
-            method,
-            params: buildMockParams({ blockParamIndex, blockParam: '0x100' }),
-          };
+      describe.each(
+        [
+          ['less than the current block number', '0x200'],
+          ['equal to the curent block number', '0x100'],
```

Comment:
is this related to retryOnEmpty or is this a new test that is unrelated?

Reply:
Yes, all of these changes are testing functionality in `retryOnEmpty`. The retries covered in this section are the ones referred to in the name "retryOnEmpty".